src/server/routes/search.js

```diff
@@ -1,10 +1,84 @@
+const algoliasearch = require('algoliasearch');
+const _ = require('lodash');
+
+const hitsPerPage = 5;
+const applicationId = '0R8E5SPAFP';
+const apiKey = '2ee58f908e6d720c4f72a9645ce0e857'; // search-only API key
+const client = algoliasearch(applicationId, apiKey);
+const index = client.initIndex('sonarwhal');
+
+const isEqual = (l, r) => {
+    // hits that have the same category, title and subtitles are considered the same
+    const checks = ['category', 'title', 'subtitle'];
+
+    return checks.every((check) => {
+        return l[check] === r[check];
+    });
+};
+
+const search = async (query, page) => {
+    let content;
+    const searchOptions = {
+        highlightPostTag: `</span>`,
+        highlightPreTag: `<span class="search-result__highlighted">`,
+        hitsPerPage,
+        page,
+        query,
+        snippetEllipsisText: `…`
+    };
+
+    try {
+        content = await index.search(searchOptions);
+    } catch (err) {
+        console.error(err);
+
+        return;
+    }
+
+    const hits = content.hits;
+    const filteredHits = hits.map((hit) => {
```

Comment:
We are doing a `map` here so the name `filteredHits` doesn't have too much sense IMO.